internal/client/blocks.go

```diff
@@ -0,0 +1,68 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.BlockTypeClient(&BlockTypeClient{})
+
+// BlockTypeClient is a client for working with block types.
+type BlockTypeClient struct {
+	hc          *http.Client
+	routePrefix string
+	apiKey      string
+}
+
+// BlockTypes returns a BlockTypeClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) BlockTypes(accountID uuid.UUID, workspaceID uuid.UUID) (api.BlockTypeClient, error) {
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	return &BlockTypeClient{
+		hc:          c.hc,
+		apiKey:      c.apiKey,
+		routePrefix: getWorkspaceScopedURL(c.endpoint, accountID, workspaceID, "block_types"),
+	}, nil
+}
```

Comment:
one quick thing - for every new client factory, we'll want to add add the factory's function signature into the `PrefectClient` interface here:

https://github.com/PrefectHQ/terraform-provider-prefect/blob/main/internal/api/client.go#L8-L20

that way, we can have our client types for when we instantiate the actual client -> pass into our provider at boot time

https://github.com/PrefectHQ/terraform-provider-prefect/blob/main/internal/provider/provider.go#L194-L198

Reply:
Makes sense! Updated in 88e3bb6.